subprojects/core/src/main/java/org/gradle/api/internal/artifacts/dependencies/DefaultProjectDependency.java

```diff
@@ -81,7 +81,7 @@ public Configuration findProjectConfiguration() {
         String declaredConfiguration = getTargetConfiguration();
         Configuration selectedConfiguration = dependencyConfigurations.getByName(GUtil.elvis(declaredConfiguration, Dependency.DEFAULT_CONFIGURATION));
         if (!selectedConfiguration.isCanBeConsumed()) {
-            throw new ConfigurationNotConsumableException(selectedConfiguration.getName());
+            throw new ConfigurationNotConsumableException(dependencyProject.toString(), selectedConfiguration.getName());
```

Comment:
Perhaps use the selected configuration's `displayName` here, instead of recalculating it from project's `toString()` and the configuration's `name`. This way, it will be consistent with how the configuration is referenced in other places.